src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,331 @@
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  // Token Symbols
+  const toTokenSymbol = coreTokens.find((token) => token.address === toToken)?.symbol
+  const fromTokenSymbol = coreTokens.find((token) => token.address === fromToken)?.symbol
+
+  useEffect(() => {
+    if (fetchSwapQuoteError) {
+      dispatch(showError(ErrorMessages.FETCH_SWAP_QUOTE_FAILED))
+      setTimeout(() => {
+        navigateBack()
+      }, 300)
+    }
+  }, [fetchSwapQuoteError])
+
+  useEffect(() => {
+    ValoraAnalytics.track(SwapEvents.swap_review_screen_open, {
+      toToken,
+      fromToken,
+      swapAmount,
+    })
+
+    void getSwapInfo().then((data) => {
+      setSwapInfo(data)
+    })
+  }, [])
+
+  const getSwapInfo = async () => {
+    try {
+      setLoading(true)
+      const swapAmountInWei = multiplyByWei(swapAmount[updatedField]!)
+      const swapAmountParam = updatedField === Field.FROM ? 'sellAmount' : 'buyAmount'
+      const params = {
+        buyToken: toToken,
+        sellToken: fromToken,
+        [swapAmountParam]: swapAmountInWei.toString().split('.')[0],
+        slippagePercentage: maxSlippageDecimal,
+        userAddress: walletAddress,
+      }
+      const requestUrl = `${networkConfig.approveSwapUrl}?${qs.stringify(params)}`
+      const response = await fetch(requestUrl)
+      if (!response.ok) {
+        Logger.error(TAG, `Failure response fetching token Swap: ${response}`)
+        throw new Error(
+          `Failure response fetching token swap quote. ${response.status}  ${response.statusText}`
+        )
+      }
+      return await response.json()
+    } catch (error) {
+      setFetchSwapQuoteError(true)
+      Logger.error(TAG, `Error fetching token swap quote: ${error}`)
+    } finally {
+      setTimeout(() => {
+        setLoading(false)
+      }, 300)
+    }
+  }
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader
+        title={t('swapReviewScreen.title')}
+        left={<BackButton />}
+        // handle large text size to avoid overlaps in header
+        right={
+          PixelRatio.getFontScale() > 1.35 ? (
+            <TopBarIconButton
+              icon={<Times />}
+              testID="CancelButton"
+              onPress={navigateHome}
+              style={{ paddingRight: Spacing.Regular16 }}
+            />
+          ) : (
+            <TopBarTextButton
+              title={t('cancel')}
+              onPress={navigateHome}
+              titleStyle={{ color: colors.gray4 }}
+            />
+          )
+        }
+      />
+      <DisconnectBanner />
+      {loading || fetchSwapQuoteError ? (
+        <View style={styles.loadingContentContainer}>
+          <LoadingSpinner />
+        </View>
+      ) : (
+        <ScrollView style={styles.contentContainer}>
+          <View style={styles.subContentContainer}>
+            <View style={styles.tallRow}>
+              <View style={styles.column}>
```

Comment:
one quick simplification you can try, is removing the `column` style entirely (and corresponding View's) and adding `flex: 1` to the `label` and `transactionDetailsRightText` and `tokenDisplayView` styles

Reply:
Styles adjusted in https://github.com/valora-inc/wallet/pull/2767/commits/70960e345f0debe2cbefb3fce5e70ef19e3fddca!